opentracing/scope_manager.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2017 The OpenTracing Authors.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+from .span import Span, SpanContext
+from .scope import Scope
+
+
+class ScopeManager(object):
+    """The `ScopeManager` interface abstracts both the activation of `Span`
+    instances (via `ScopeManager#activate(Span)`) and access to an active
+    `Span` / `Scope` (via `ScopeManager#active()`).
+    """
+    def __init__(self):
+        # TODO: `tracer` should not be None, but we don't have a reference;
+        # should we move the NOOP SpanContext, Span, Scope to somewhere
+        # else so that they're globally reachable?
```

Comment:
What I mean here is that maybe everywhere in the repo we can simply:
```python
from opentracing.ext import noop  # or something similar

self._noop_span = noop.SPAN
self._noop_scope = noop.SCOPE
```

The `noop` module may be initialized when the `Tracer` is created (so in the `__init__.py` at the moment?) so we have references of `Tracer` and `ScopeManager`, avoiding creating every time a no-op of something.

Let me know if I'm missing something!